src/main/java/org/openstreetmap/atlas/geography/atlas/change/Change.java

```diff
@@ -253,6 +263,77 @@ public void save(final WritableResource resource, final boolean showDescription)
+
+        return builder.toString();
+    }
+
+    public Map<ItemType, Map<ChangeDescriptorType, Map<String, AtomicLong>>> tagCountMap()
+    {
+        final Map<ItemType, Map<ChangeDescriptorType, Map<String, AtomicLong>>> tagMap = new EnumMap<>(
+                ItemType.class);
+        for (final ItemType itemType : ItemType.values())
+        {
+            final Map<ChangeDescriptorType, Map<String, AtomicLong>> descriptorMap = new EnumMap<>(
+                    ChangeDescriptorType.class);
+            for (final ChangeDescriptorType type : ChangeDescriptorType.values())
+            {
+                descriptorMap.put(type, new HashMap<>());
+            }
+            tagMap.put(itemType, descriptorMap);
+            this.featureChanges.stream().filter(change -> change.getItemType().equals(itemType)
+                    && change.explain().getChangeDescriptorType()
+                            .equals(ChangeDescriptorType.UPDATE)
+                    && change.explain().getChangeDescriptors().stream().anyMatch(
+                            descriptor -> descriptor.getName().equals(ChangeDescriptorName.TAG)))
+                    .forEach(
+                            change -> change.explain().getChangeDescriptors().stream()
+                                    .filter(changeDescriptor -> changeDescriptor.getName()
+                                            .equals(ChangeDescriptorName.TAG))
+                                    .forEach(changeDescriptor ->
+                                    {
+                                        final TagChangeDescriptor tagChangeDescriptor = (TagChangeDescriptor) changeDescriptor;
+                                        if (tagMap.get(itemType)
+                                                .get(tagChangeDescriptor.getChangeDescriptorType())
+                                                .containsKey(tagChangeDescriptor.getKey()))
+                                        {
+                                            tagMap.get(itemType)
+                                                    .get(tagChangeDescriptor
+                                                            .getChangeDescriptorType())
+                                                    .get(tagChangeDescriptor.getKey())
+                                                    .incrementAndGet();
+                                        }
+                                        else
+                                        {
+                                            tagMap.get(itemType)
+                                                    .get(tagChangeDescriptor
+                                                            .getChangeDescriptorType())
+                                                    .put(tagChangeDescriptor.getKey(),
+                                                            new AtomicLong(1));
+                                        }
+                                    }));
+        }
```

Comment:
Wow. Haha

Reply:
Could we maybe just get a quick comment above this block about why it's here/what it is intended to do? That's a pretty dense cluster of stream operations haha